pkg/kubelet/stats/cadvisor_stats_provider_test.go

```diff
@@ -367,6 +367,124 @@ func TestCadvisorListPodStats(t *testing.T) {
+
+		// Pod1 - Namespace0
+		"/pod1-i":             getTestContainerInfo(seedPod1Infra, pName1, namespace0, kubelettypes.PodInfraContainerName),
+		"/pod1-c0":            getTestContainerInfo(seedPod1Container, pName1, namespace0, cName10),
+		"/pod1-c2-terminated": getTerminatedContainerInfo(seedPod1Container, pName1, namespace0, cName2),
+		//"/pod1-c2-terminated": getTestContainerInfo(seedPod1Container, pName1, namespace0, cName2),
+	}
+
+	freeRootfsInodes := rootfsInodesFree
+	totalRootfsInodes := rootfsInodes
+	rootfs := cadvisorapiv2.FsInfo{
+		Capacity:   rootfsCapacity,
+		Available:  rootfsAvailable,
+		InodesFree: &freeRootfsInodes,
+		Inodes:     &totalRootfsInodes,
+	}
+
+	freeImagefsInodes := imagefsInodesFree
+	totalImagefsInodes := imagefsInodes
+	imagefs := cadvisorapiv2.FsInfo{
+		Capacity:   imagefsCapacity,
+		Available:  imagefsAvailable,
+		InodesFree: &freeImagefsInodes,
+		Inodes:     &totalImagefsInodes,
+	}
+
+	options := cadvisorapiv2.RequestOptions{
+		IdType:    cadvisorapiv2.TypeName,
+		Count:     2,
+		Recursive: true,
+	}
+
+	mockCadvisor := cadvisortest.NewMockInterface(t)
+	mockCadvisor.EXPECT().ContainerInfoV2("/", options).Return(infos, nil)
+	mockCadvisor.EXPECT().RootFsInfo().Return(rootfs, nil)
+	mockCadvisor.EXPECT().ImagesFsInfo(ctx).Return(imagefs, nil)
+
+	mockRuntime := containertest.NewMockRuntime(t)
+
+	ephemeralVolumes := []statsapi.VolumeStats{getPodVolumeStats(seedEphemeralVolume1, "ephemeralVolume1"),
+		getPodVolumeStats(seedEphemeralVolume2, "ephemeralVolume2")}
+	persistentVolumes := []statsapi.VolumeStats{getPodVolumeStats(seedPersistentVolume1, "persistentVolume1"),
+		getPodVolumeStats(seedPersistentVolume2, "persistentVolume2")}
+	volumeStats := serverstats.PodVolumeStats{
+		EphemeralVolumes:  ephemeralVolumes,
+		PersistentVolumes: persistentVolumes,
+	}
+	p1Time := metav1.Now()
+	mockStatus := statustest.NewMockPodStatusProvider(t)
+	mockStatus.EXPECT().GetPodStatus(types.UID("UID"+pName1)).Return(v1.PodStatus{StartTime: &p1Time}, true)
+
+	resourceAnalyzer := &fakeResourceAnalyzer{podVolumeStats: volumeStats}
+
+	p := NewCadvisorStatsProvider(mockCadvisor, resourceAnalyzer, nil, nil, mockRuntime, mockStatus, NewFakeHostStatsProvider())
+	pods, err := p.ListPodStats(ctx)
+	assert.NoError(t, err)
+
+	assert.Len(t, pods, 1)
+	indexPods := make(map[statsapi.PodReference]statsapi.PodStats, len(pods))
+	for _, pod := range pods {
+		indexPods[pod.PodRef] = pod
+	}
+
+	// Validate Pod1 Results
+	ps, found := indexPods[prf1]
+	assert.True(t, found)
+	assert.Len(t, ps.Containers, 2)
+	con := ps.Containers[0]
+	assert.Equal(t, cName10, con.Name)
+	checkCPUStats(t, "Pod1Container0", seedPod1Container, con.CPU)
+	checkMemoryStats(t, "Pod1Container0", seedPod1Container, infos["/pod1-c0"], con.Memory)
+	checkSwapStats(t, "Pod1Container0", seedPod1Container, infos["/pod1-c0"], con.Swap)
+	checkNetworkStats(t, "Pod1", seedPod1Infra, ps.Network)
```

Comment:
IIUC this currently tests that 2 containers are being considered, even though one of them is "dead', right?
Do we want to check the precise storage usage?
Do we want to add an e2e test?